src/hooks/queries/useSafeTx.ts

```diff
@@ -1,22 +1,29 @@
-import type { ChainId } from '@shapeshiftoss/caip'
+import type { AccountId } from '@shapeshiftoss/caip'
 import { fetchSafeTransactionInfo } from '@shapeshiftoss/swapper/dist/safe-utils'
 import { type SafeTxInfo } from '@shapeshiftoss/swapper/dist/safe-utils/types'
 import type { UseQueryResult } from '@tanstack/react-query'
 import { skipToken, useQuery } from '@tanstack/react-query'
+import { fetchIsSmartContractAddressQuery } from 'hooks/useIsSmartContractAddress/useIsSmartContractAddress'
 
 type UseSafeTxQueryArgs = {
-  chainId: ChainId
+  accountId: AccountId | undefined
```

Comment:
The guts of this PR and its plumbing: we now pass an `AccountId` to `useSafeTxQuery` and consume SAFE TX data holistically.

The reason why we do said plumbing is so that we can do preliminary checks in `fetchSafeTransactionInfo` re: is the current address a SAFE account (with an infinitely cached query, passing the curried `queryFn` leveraging web provider up-top so we don't overfetch), and early bail if so, so that the app doesn't spew requests to SAFE API that we don't need.